autoscheduler/scraper/management/commands/scrape_grades.py

```diff
@@ -0,0 +1,259 @@
+
+        try:
+            # Throws a ObjectDoesNotExist error upon failure
+            section = Section.objects.get(
+                subject=dept,
+                course_num=grade_data.course_num,
+                section_num=grade_data.section_num,
+                term_code=int(term)
+            )
+
+            # Increment how many grades for this department were scraped
+            count = counts.get(grade_data.dept) or 0
+            counts[grade_data.dept] = count + 1
+
+            # Create the grade model and add it to the list of models to be returned
+            grade = Grades(section=section, gpa=grade_data.gpa,
+                           **grade_data.letter_grades)
+
+            scraped_grades.append(grade)
+
+        except ObjectDoesNotExist:
+            print((f"Section couldn't be found for {term} {grade_data.dept}-"
+                   f"{grade_data.course_num} {grade_data.section_num}"))
+
+    if len(counts.items()) == 0:
+        print(f"No grades scraped")
+
+    # Not sure if we want this, maybe just with the debug flag?
+    for (dept, count) in counts.items():
+        print(f"{dept}: Scraped {count} grades")
+    print() # Add a new line to separate the outputs
+
+    return scraped_grades
+
+async def perform_searches(years: List[int], colleges: List[str]) -> List[Grades]:
+    """ Gathers all of the retrieve_pdf tasks for each year/semester/college combination
+
+        Then collects the returned list of grade models and returns them
+     """
+
+    loop = asyncio.get_running_loop()
+
+    async def retrieve_pdf(year_semester: str, college: str):
+        """ Creates a aiohttp session, downloads the grade report pdf, parses & scrapes it
+            and returns the list of scraped grades
+        """
+
+        async with ClientSession(loop=loop) as session:
+            pdf_path = await download_pdf(year_semester, college, session)
+
+            if pdf_path is not None:
+                # Parse the pdf then scrape the returned grade distributions
+                print(f"Parsing PDF for {year_semester} {college}")
+
+                grade_dists = pdf_parser.parse_pdf(pdf_path)
+
+                # Convert the year_semester term to include the location
+                term = generate_term_with_location(year_semester, college)
+
+                scraped_grades = scrape_pdf(grade_dists, term)
+
+                return scraped_grades
+
+            return None
+
+    semesters = [SPRING, SUMMER, FALL]
+
+    # Gather all of the retrieve functions into an array
+    tasks = []
+    for year in years:
+        for semester in semesters:
+            for college in colleges:
+                year_semester_term = str(year) + semester
+                tasks.append(retrieve_pdf(year_semester_term, college))
+
+    grades = [] # list of Grades models
+    for scraped_grades in await asyncio.gather(*tasks, loop=loop):
+        if scraped_grades is not None and len(scraped_grades) > 0:

```

Comment:
```suggestion
        if scraped_grades:
```
Handles both cases at once